v3.2/glfw/util.go

```diff
@@ -21,3 +23,23 @@ func bytes(origin []byte) (pointer *uint8, free func()) {
 	ptr := C.CBytes(origin)
 	return (*uint8)(ptr), func() { C.free(ptr) }
 }
+
+func imageToGLFW(img image.Image) (r C.GLFWimage, free func()) {
+	b := img.Bounds()
+
+	r.width = C.int(b.Dx())
+	r.height = C.int(b.Dy())
+
+	if m, ok := img.(*image.NRGBA); ok && m.Stride == b.Dx()*4 && b.Min.X == 0 {
+		pix, free := bytes(m.Pix[b.Min.Y*m.Stride : b.Max.Y*m.Stride])
```

Comment:
Does this really work when `b.Min.Y` and/or `b.Max.Y` is negative? I thought just `m.Pix` should work.

https://pkg.go.dev/image#NRGBA

> // Pix holds the image's pixels, in R, G, B, A order. The pixel at
> // (x, y) starts at Pix[(y-Rect.Min.Y)*Stride + (x-Rect.Min.X)*4].
> Pix [][uint8](https://pkg.go.dev/builtin#uint8)

Reply:
Yeah, you seem right. For some reason, I was thinking that it didn't subslice the pix when making a subimage.

I guess the better approach would be:

```
if m, ok := img.(*image.NRGBA); ok && m.Stride == b.Dx()*4 {
	start := m.PixOffset(m.Rect.Min.X, m.Rect.Min.Y)
	end := m.PixOffset(m.Rect.Max.X, m.Rect.Max.Y)
	pix, free := bytes(m.Pix[start:end])
```

Then it's clear that the ranges are calculated correctly.

Reply:
So, yeah, start ends up always `0`, however, end might be something different, because Subimage doesn't limit the length https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/image/image.go;l=430